api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -336,6 +338,20 @@ private void formatTimestampValues(List<DataSetPreviewValueList> valuePreviewLis
             qualifier,
             queriesByDomain);
 
+    if (GenomicsDataTypeEnum.MICROARRAY.equals(dataSetExportRequest.getGenomicsDataType())) {
+      if (!dataSetExportRequest.getKernelType().equals(KernelTypeEnum.PYTHON)) {
```

Comment:
If it's easy to check the CDR version (for the workspace, e.g. workspace.getCdrVersion()) here - an assertion about whether this workspace has genomics data could be a nice addition as well. I'd consider that failure mode to be a 412 failed precondition since it's dependent on CDR version state.